lib/iterators.js

```diff
@@ -61,19 +67,35 @@ function traverseSchema(schema, callback, prop, schemaTypesToIterate) {
   if (!schemaTypesToIterate.includes(SchemaTypesToIterate.objects) && schema.type() === 'object') return;
   if (schema.isCircular()) return;
   if (callback(schema, prop, SchemaIteratorCallbackType.NEW_SCHEMA) === false) return;
-  
-  if (schema.type() !== undefined) {
-    switch (schema.type()) {
-    case 'object':
-      recursiveSchemaObject(schema, callback, schemaTypesToIterate);
-      break;
-    case 'array':
-      recursiveSchemaArray(schema, callback, schemaTypesToIterate);
-      break;
+  if (schema.patternProperties() !== undefined && schemaTypesToIterate.includes(SchemaTypesToIterate.patternProperties)) {
```

Comment:
Shall I make functions like `traversePatternProperties()`, `traverseDependencies()`, `traversePatterndefinitions()`, and wrap the code there because linter tests are not passing due to Cognitive Complexity?

Reply:
🤔 I think we could reduce complexity by providing some helper functions such as https://github.com/asyncapi/parser-js/blob/7bb3aaf2452f336775e1cdd56c290976c25ffcde/lib/iterators.js#L112 but to iterate JS objects. As you can see it is the same code that is being run just with different objects.

Wanna try create such a function? 🙂

Reply:
Yeah, On it!